cli/pcluster/easyconfig.py

```diff
@@ -305,6 +296,73 @@ def _is_config_valid(args, config):
         return False
 
 
+def get_parameter(config, section, parameter_name, default_value):
+    """
+    Prompt the user to ask question without validation
+    :param config the configuration parser
+    :param section the name of the section
+    :param parameter_name: the name of the parameter
+    :param default_value: the default string to ask the user
+    :return:
+    """
+    return config.get(section, parameter_name) if config.has_option(section, parameter_name) else default_value
+
+
+def general_scheduler_handler(config, cluster_template):
+    """
+    Return a dictionary containing the values asked to the user for a generic scheduler non aws_batch
+    :param config the configuration parser
+    :param cluster_template the name of the cluster
+    :return: a dictionary with the updated values
+    """
+    scheduler_dict = {}
+
+    # We first remove unnecessary parameters from the past configurations
+    batch_parameters = "max_vcpus", "desired_vcpus", "min_vcpus"
+    for par in batch_parameters:
+        config.remove_option("cluster " + cluster_template, par)
+
+    operating_system = prompt(
+        "Operating System",
+        get_parameter(config, "cluster " + cluster_template, "base_os", "alinux"),
+        options=get_supported_os(),
+        check_validity=True,
+    )
+    scheduler_dict["base_os"] = operating_system
+
+    max_queue_size = prompt(
+        "Max Queue Size",
+        get_parameter(config, "cluster " + cluster_template, "max_queue_size", "10")
+    )
+
+    scheduler_dict["max_queue_size"] = max_queue_size
+    scheduler_dict["initial_queue_size"] = "1"
+    return scheduler_dict
+
+
+def aws_batch_handler(config, cluster_template):
```

Comment:
We are modifying the `config` object so I'd rename the function as: `_init_awsbatch_params` or see the comment below for `_init_scaling_params(..., scheduler)`